app/controllers/blogs_controller.rb

```diff
@@ -53,53 +50,49 @@ def edit
         redirect_to root_path
       end
     else
-      flash[:alert] = "Please Login First"
-      redirect_to sign_in_path
+      log_in
     end
   end
 
-    def update
-      if Current.user
-        @blog = Current.user.blogs.where(id: params[:id]).first
-        if !@blog.nil? && @blog.update(blog_params)
-          flash[:notice] = "Successfully Updated"
-          redirect_to @blog
-        elsif @blog
-          flash[:alert] = "An Error Occured while Updating. Please retry!!"
-          render :edit
-        else
-          flash[:alert] = "No Blog Found"
-          redirect_to root_path
-        end
+  def update
+    if Current.user
+      @blog = Current.user.blogs.where(id: params[:id]).first
+      if !@blog.nil? && @blog.update(blog_params)
+        flash[:notice] = "Successfully Updated"
+        redirect_to @blog
+      elsif @blog
+        flash[:alert] = "An Error Occured while Updating. Please retry!!"
+        render :edit
       else
-        flash[:alert] = "Please Login First"
-        redirect_to sign_in_path
+        flash[:alert] = "No Blog Found"
+        redirect_to root_path
       end
+    else
+      log_in
     end
+  end
 
-
-    def destroy
-      if Current.user
-        @blog = Current.user.blogs.where(id: params[:id]).first
-        if !@blog.nil? && @blog.destroy
-          flash[:notice] = "Successfully Deleted"
-          redirect_to blogs_path
-        elsif @blog
-          flash[:alert] = "An error Occured. Please retry !! "
-          redirect_to @blog
-        else
-          flash[:alert] = "No blog found"
-          redirect_to root_path 
-        end
+  def destroy
+    if Current.user
+      @blog = Current.user.blogs.where(id: params[:id]).first
```

Comment:
You can also do query directly on blog model, looks cleaner
`Blog.where(:id => params[:id], :user_id => current_user.id).first`